serve/mlc_serve/model/paged_cache_model.py

```diff
@@ -607,16 +622,50 @@ def generate(
         torch.cuda.synchronize()
         torch.cuda.nvtx.range_pop()
 
-        next_tokens = sample(logits, sampling_params, self.vocab_size)
+        try:
+            next_tokens = sample(logits, sampling_params, self.vocab_size)
 
-        return [
-            TextGenerationResult(
-                sequence_id=sequence_id,
-                generated_tokens=[new_token],
-                error=None,
-            )
-            for sequence_id, new_token in zip(sequence_ids, next_tokens)
-        ]
+            return [
+                TextGenerationResult(
+                    sequence_id=sequence_id,
+                    generated_tokens=[new_token],
+                    error=None,
+                )
+                for sequence_id, new_token in zip(sequence_ids, next_tokens)
+            ]
+        except RuntimeError:
+            # Fallback to per-token sampling in case some logits values are corrupted.
+            outputs = []
```

Comment:
Do not fully understand why we need this section if we add `_is_safe_to_sample` and verify logits in sample. We must not be here

Reply:
Why do we need to verify on the second try? If it is a problem with nan/<0 values, this verificatin will not change anything, second sample() call will fail as well. But if it is another error, we will get another exception in except block.

Reply:
The first sample is over batched tokens. We skip verification on the first sample to prevent perf regression and the error is rare anyway. After the error is raised, we don't know which token's logits is corrupted. So the second sample is run over each token sequentially. Check the code under `except RuntimeError:` for details.